### PR TITLE
docs: extended the flag configuration documentation

### LINKED ISSUE #680: [DOC] Improve the flag configuration docs
### Change in the documentation

Expand the flag configuration docs to explain the full configuration schema.

### PR DESCRIPTION
**Pull request by beeme1mr**
# This PR

- Includes a description for the full flag configuration
- Expands on the targeting rule conditions, properties, and functions.

### Related Issues

Fixes #680

### CODE REVIEW

File: docs/configuration/flag_configuration.md [hunk 1]
```diff
@@ -103,43 +162,133 @@ The output of the targeting rule **must** match the name of one of the variants
...
-      ]
-    }
-    ```
+#### Evaluation Context
+
+Evaluation context can be accessed in targeting rules using the `var` property followed the evaluation context property name.
+
+| Description                                                        | Example                                    |
+| ------------------------------------------------------------------ | ------------------------------------------ |
+| Retrieve property from the evaluation context                      | { "var": "email" }                         |
```
Comment (toddbaert) [inline docs/configuration/flag_configuration.md:171]: examples here arent monospaced.

File: docs/configuration/flag_configuration.md [hunk 2]
```diff
@@ -103,43 +162,133 @@ The output of the targeting rule **must** match the name of one of the variants
...
+| Retrieve property from the evaluation context or use default value | { "var": ["email", "noreply@example.com] } |
+| Retrieve a nested property from the evaluation context             | { "var": "user.email" }                    |
+
+> For more information, see the `var` section in the [JSON Logic documentation](https://jsonlogic.com/operations.html#var).
+
+#### Conditions
+
+| Conditional | Example                                                                                                                                                                                |
+| ----------- | -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| If          | Logic: `{"if" : [ true, "yes", "no" ]}`<br>Result: "yes"<br><br>Logic: `{"if" : [ false, "yes", "no" ]}`<br>Result: "no"                                                               |
```
Comment (toddbaert) [inline docs/configuration/flag_configuration.md:181]: consider putting the results in `monospace`.

File: docs/configuration/flag_configuration.md [hunk 3]
```diff
@@ -103,43 +162,133 @@ The output of the targeting rule **must** match the name of one of the variants
...
+| Not it                 | Attribute is not in an array of strings                              | string       | Logic: `{ "!": { "in" : [ "Mike", ["Bob", "Mike"]] } }`<br>Result: false<br><br>Logic: `{ "!": { "in":["Todd", ["Bob", "Mike"]] } }`<br>Result: true |
+
+### Functions
+
+| Conditional           | Description                               | Example                                                                                                                                                                      |
+| --------------------- | ----------------------------------------- | ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| Fractional evaluation | A deterministic percentage-based rollout  | A complete example can be found [here](./reusable_targeting_rules.md)                                                                                                        |
```
Comment (toddbaert) [inline docs/configuration/flag_configuration.md:211]: I think this link is wrong.
```suggestion
| Fractional evaluation | A deterministic percentage-based rollout  | A complete example can be found [here](./fractional_evaluation.md)                                                                                                        |
```

**Review by toddbaert — APPROVED** [review]
Approved assuming you fix this: https://github.com/open-feature/flagd/pull/681/files#r1212095517

Added some other nits.

File: docs/configuration/flag_configuration.md [hunk 4]
```diff
@@ -103,43 +162,133 @@ The output of the targeting rule **must** match the name of one of the variants
...
-              "var": ["email"]
-            }
-          ]
-        },
-        "binet",
-        null
-      ]
-    }
-    ```
+#### Evaluation Context
```
Comment (Kavindu-Dodan) [inline docs/configuration/flag_configuration.md:165]: I think we should explain what evaluation context means. Highlighting that this comes from flag evaluation request body. And also provide an example to connect with variables defined below, 

```
{
    "flagKey":"booleanFlagKey",
    "context":{
        "email": "abc@abc.com"
    }
}
```

File: docs/configuration/flag_configuration.md [hunk 5]
```diff
@@ -103,43 +162,133 @@ The output of the targeting rule **must** match the name of one of the variants
...
+
+Evaluation context can be accessed in targeting rules using the `var` property followed the evaluation context property name.
+
+| Description                                                        | Example                                    |
+| ------------------------------------------------------------------ | ------------------------------------------ |
+| Retrieve property from the evaluation context                      | { "var": "email" }                         |
+| Retrieve property from the evaluation context or use default value | { "var": ["email", "noreply@example.com] } |
```
Comment (Kavindu-Dodan) [inline docs/configuration/flag_configuration.md:172]: Nit - 
```suggestion
| Retrieve property from the evaluation context or use default value | { "var": ["email", "noreply@example.com"] } |
```

File: docs/configuration/flag_configuration.md [hunk 6]
```diff
@@ -103,43 +162,133 @@ The output of the targeting rule **must** match the name of one of the variants
...
+| Contains               | Contains string                                                      | string       | Logic: `{ "in": ["Spring", "Springfield"] }`<br>Result: true<br><br>Logic: `{ "in":["Illinois", "Springfield"] }`<br>Result: false                   |
+| Not contains           | Does not contain a string                                            | string       | Logic: `{ "!": { "in":["Spring", "Springfield"] } }`<br>Result: false<br><br>Logic: `{ "!": { "in":["Illinois", "Springfield"] } }`<br>Result: true  |
+| In                     | Attribute is in an array of strings                                  | string       | Logic: `{ "in" : [ "Mike", ["Bob", "Mike"]] }`<br>Result: true<br><br>Logic: `{ "in":["Todd", ["Bob", "Mike"]] }`<br>Result: false                   |
+| Not it                 | Attribute is not in an array of strings                              | string       | Logic: `{ "!": { "in" : [ "Mike", ["Bob", "Mike"]] } }`<br>Result: false<br><br>Logic: `{ "!": { "in":["Todd", ["Bob", "Mike"]] } }`<br>Result: true |
+
+### Functions
+
```
Comment (Kavindu-Dodan) [inline docs/configuration/flag_configuration.md:208]: Shall we mention the fact that these are special flagd implemenation?

**Review by Kavindu-Dodan — APPROVED** [review]
Left minor remarks, otherwise LGTM

File: docs/configuration/flag_configuration.md [hunk 7]
```diff
@@ -100,46 +159,149 @@ Example of an invalid configuration:
...
-              "var": ["email"]
-            }
-          ]
-        },
-        "binet",
-        null
-      ]
-    }
-    ```
+The evaluation context can be accessed in targeting rules using the `var` property followed the evaluation context property name.
```
Comment (toddbaert) [inline docs/configuration/flag_configuration.md:179]:
```suggestion
The evaluation context can be accessed in targeting rules using the `var` operation followed the evaluation context property name.
```

You could go with `keyword` or just say "by using `var`" - but I think "property" is wrong here.

**Review by toddbaert — APPROVED** [review]
Reviewed changes since yesterday, LGTM.

**Review by toddbaert — APPROVED** [review]
Reviewed changes since yesterday, LGTM.

**Pull request merged**